### PR TITLE
docs: update PR template to include a checklist

### LINKED ISSUE #4867: Docs: Update PR template to include a checklist for authors and maintainers
In the past couple of weeks, we've received a few community contributions to this repo. AFAIK, InfluxData requires non-employee contributors to sign the [CLA](https://www.influxdata.com/legal/cla/), to protect the company from liability. This isn't mentioned anywhere in the ui repo, even though it's an open-source project.

### Acceptance Criteria
- Bring the PR template in line with other repos in our organization
- Include a checklist in the PR template that includes signing the CLA

### PR DESCRIPTION
**Pull request by hoorayimhelping**
Closes #4867

Adds a PR template with a basic checklist

Let's get meta, here's what it'll look like:

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- ~[ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)~
- ~[ ] Feature flagged, if applicable~

### CODE REVIEW

**Review by wdoconnell — APPROVED** [review]
LGTM. 👍

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
 Closes #
 
 <!-- Describe your proposed changes here. -->
+<!-- A picture is worth a thousand words. -->
```
[the remark this answers is not in the document]
Comment (wdoconnell) [reply .github/pull_request_template.md:4]: If we prefer pictures, maybe we put this in the instructions/checklist? Folks might just ignore/delete if it's in a comment.

[the remark this answers is not in the document]
Comment (hoorayimhelping) [reply .github/pull_request_template.md:4]: I meant it as a reminder to include screenshots. I'll update it since it doesn't seem clear

**Comment by randycoulman** [comment]
I don't work in this repo enough to have a vote one way or the other. But since I was asked for a review, I'll share some thoughts.

I tend to agree with @rockstar that checklists can end up being noise that people ignore or blindly check off without thinking about it and that adding tooling to automatically enforce the important things is a good idea.

That said, we adopted [a checklist in the (closed source) Quartz repo](https://github.com/influxdata/quartz/blob/master/.github/pull_request_template.md) to remind us about things that we care about but haven't yet (fully) automated. Since it's a closed source repo, this is really just for our team (and occasionally other Influxers who contribute) and it's there to remind us of some things that we've agreed are important to us.

An option could be to decide to do a timeboxed experiment with the checklist and then discuss the results after the experiment is done. Is it helping our teams? Is it just annoying and people aren't paying attention? Is it resulting in community PRs that require less back-and-forth with reviewers? plus whatever other criteria are important.

Having a timebox with a scheduled review will hopefully make it easy to get rid of it if it doesn't help.

**Comment by rockstar** [comment]
> An option could be to decide to do a timeboxed experiment with the checklist and then discuss the results after the experiment is done. Is it helping our teams? Is it just annoying and people aren't paying attention? Is it resulting in community PRs that require less back-and-forth with reviewers? plus whatever other criteria are important.

I. Frickin. Love. This. Idea.

Sometimes what we claim is important to us doesn't actually become important, and so it's a way for us to check in on those commitments. Maybe we get through the experiment and prioritize some work to automate it, because it's important, but we can't budget the cognitive load to adhere to these but they are still important (this is, quite literally, my SOP). That might be just the kind of catalyst that put me on board for this kind of patch, rather than just shrugging and saying "It doesn't hurt, I guess."

**Comment by hoorayimhelping** [comment]
@rockstar @randycoulman Kristina created a ticket https://github.com/influxdata/ui/issues/4925 and added a note in our sprint board to it that says do this ticket in about 3 months, around October. So when we do our sprint planning, we'll see the note to deal with that ticket in the future, and when the time comes, us three can get together and figure out if this checklist is worth keeping around.

**Pull request merged**